index.js

```diff
@@ -81,7 +81,9 @@ function markings(strings /*: Array<string> */, ...values /*: Array<ReactNode> *
     renderers: {
       Paragraph: function(props) {
         if (props.children.length === 1 && props.children[0] === PLACEHOLDER) {
-          return values.shift();
+          var value = values.shift();
+          values.push(value);
+          return value;
```

Comment:
What about this?

```js
var index = 0;
// ...
var value = values[index];
index = index < values.length - 1 ? index + 1 : 0;

```

Reply:
Hmm, better, at least it doesn't alter the passed in "values". Will do it this way. Was thinking of cloning the "values" and then shift/push on the cloned value. Yours seem easier.